File: common/protcol/c#/caller/dbproxy_call_hubcaller.cs
/*this caller file is codegen by juggle for c#*/
using System;
using System.Collections;
using System.IO;

namespace caller
{
    public class dbproxy_call_hub : juggle.Icaller 
    {
        public dbproxy_call_hub(juggle.Ichannel _ch) : base(_ch)
        {
            module_name = "dbproxy_call_hub";
        }

        public void reg_hub_sucess()
        {
            ArrayList _argv = new ArrayList();
            call_module_method("reg_hub_sucess", _argv);
        }

        public void ack_create_persisted_object(String argv0)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            call_module_method("ack_create_persisted_object", _argv);
        }

        public void ack_updata_persisted_object(String argv0)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            call_module_method("ack_updata_persisted_object", _argv);
        }

        public void ack_get_object_count(String argv0,Int64 argv1)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            call_module_method("ack_get_object_count", _argv);
        }

        public void ack_get_object_info(String argv0,ArrayList argv1)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            call_module_method("ack_get_object_info", _argv);
        }

        public void ack_get_object_info_end(String argv0)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            call_module_method("ack_get_object_info_end", _argv);
        }

        public void ack_remove_object(String argv0)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            call_module_method("ack_remove_object", _argv);
        }

    }
}

File: common/protcol/js/caller/dbproxy_call_hubcaller.js
/*this caller file is codegen by juggle for js*/
function dbproxy_call_hub_caller(ch){
    Icaller.call(this, "dbproxy_call_hub", ch);

    this.reg_hub_sucess = function(){
        var _argv = [];
        this.call_module_method.call(this, "reg_hub_sucess", _argv);
    }

    this.ack_create_persisted_object = function( argv0, argv1){
        var _argv = [argv0,argv1];
        this.call_module_method.call(this, "ack_create_persisted_object", _argv);
    }

    this.ack_updata_persisted_object = function( argv0){
        var _argv = [argv0];
        this.call_module_method.call(this, "ack_updata_persisted_object", _argv);
    }

    this.ack_get_object_count = function( argv0, argv1){
        var _argv = [argv0,argv1];
        this.call_module_method.call(this, "ack_get_object_count", _argv);
    }

    this.ack_get_object_info = function( argv0, argv1){
        var _argv = [argv0,argv1];
        this.call_module_method.call(this, "ack_get_object_info", _argv);
    }

    this.ack_get_object_info_end = function( argv0){
        var _argv = [argv0];
        this.call_module_method.call(this, "ack_get_object_info_end", _argv);
    }

    this.ack_remove_object = function( argv0){
        var _argv = [argv0];
        this.call_module_method.call(this, "ack_remove_object", _argv);
    }

}
(function(){
    var Super = function(){};
    Super.prototype = Icaller.prototype;
    dbproxy_call_hub_caller.prototype = new Super();
})();
dbproxy_call_hub_caller.prototype.constructor = dbproxy_call_hub_caller;

